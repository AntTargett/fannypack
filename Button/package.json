{
  "name": "fannypack/Button",
  "private": true,
  "main": "../lib/Button/index.js",
  "module": "../es/Button/index.js"
}