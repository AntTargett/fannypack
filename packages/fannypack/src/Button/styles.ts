import { css, cssClass, palette, space, theme } from '../styled';

export const Button = styleProps => cssClass`
  align-items: center;
  background-color: ${palette()(styleProps)};
  border-radius: 4px;
  color: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
  fill: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
  cursor: pointer;
  display: inline-flex;
  font-weight: ${theme('fontWeights.semibold')(styleProps)};
  min-height: 2.5em;
  justify-content: center;
  padding: 0 0.8rem;
  text-decoration: none;
  hyphens: auto;

  ${styleProps.palette === 'default' &&
    css`
      border: 1px solid ${palette('gray100')(styleProps)};
    `};

  & {
    ${theme('Button.base')(styleProps)};
  }

  &:focus {
    outline: unset;
    z-index: 2;
    box-shadow: ${palette(styleProps.palette === 'default' ? 'primary' : `${styleProps.palette}300`)(
      styleProps
    )} 0px 0px 0px
      2px;

    ${theme('Button.focus')(styleProps)};
  }

  ${styleProps.size && getSizeProperties(styleProps)}
`;

export const getSizeProperties = styleProps => {
  const styles = {
    small: css`
      & {
        font-size: ${theme('fontSizes.100')(styleProps)}em;
        min-height: ${space(8)(styleProps)}em;
        padding: 0 ${space(2)(styleProps)}rem;
      }
      & {
        ${theme('Button.sizes.small')(styleProps)};
      }
    `,
    default: css`
      & {
        ${theme('Button.sizes.default')(styleProps)};
      }
    `,
    medium: css`
      & {
        min-height: ${space(12)(styleProps)}em;
        padding: 0 ${space(5)(styleProps)}rem;
      }
      & {
        ${theme('Button.sizes.medium')(styleProps)};
      }
    `,
    large: css`
      & {
        font-size: ${theme('fontSizes.300')(styleProps)}em;
        min-height: ${space(12)(styleProps)}em;
        padding: 0 ${space(6)(styleProps)}rem;
      }
      & {
        ${theme('Button.sizes.large')(styleProps)};
      }
    `
  };
  return styles[styleProps.size];
};
