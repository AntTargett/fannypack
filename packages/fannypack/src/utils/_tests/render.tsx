import * as React from 'react';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (Component: any, { theme = {} } = {}) => render(<Provider theme={theme}>{Component}</Provider>);
