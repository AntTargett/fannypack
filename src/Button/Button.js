import React from 'react';
import PropTypes from 'prop-types';
import * as sharedPropTypes from '../prop-types';

import { Button as StyledButton } from './styled';

const Button = ({ children, color }) => <StyledButton color={color}>{children}</StyledButton>;

Button.propTypes = {
  children: PropTypes.node.isRequired,
  /** Color of the button. Available values: "default", "primary", "secondary", "success", "warning", "danger" */
  color: sharedPropTypes.color
};

Button.defaultProps = {
  color: 'default'
};

Button.propertyControls = {
  children: { type: 'string', title: 'Text' },
  color: { type: 'enum', options: ['default', 'primary', 'success', 'danger'], title: 'Color' }
};

export default Button;
