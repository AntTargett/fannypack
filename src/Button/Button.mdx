---
name: Button
route: /elements/button
menu: Elements
---

import { Playground, PropsTable } from 'docz';
import { Button, Buttons } from '../index';

## Import

`import { Button, Buttons } from 'bumbag'`

## Basic Usage

<Playground>
  <Button>Default button</Button>
</Playground>

## Custom elements

Button, by default, is actually wrapped in a `div` element. See more [here](https://reakit.io/components/button).
However, you are able to change it to a custom element using the `as` prop.

<Playground>
  <Buttons>
    <Button>Default</Button>
    <Button as="a" href="#">Anchor</Button>
  </Buttons>
</Playground>

## Colors

Colors can be added to a button with the `color` prop.

<Playground>
  <Buttons>
    <Button>Default</Button>
    <Button state="primary">Primary</Button>
    <Button state="success">Success</Button>
    <Button state="danger">Danger</Button>
    <Button state="warning">Warning</Button>
  </Buttons>
</Playground>

## Loading

To indicate a button is loading, use the `isLoading` prop.

<Playground>
  <Buttons>
    <Button isLoading>Default</Button>
    <Button state="primary" isLoading>Primary</Button>
    <Button state="primary" isLoading type="link">Primary</Button>
    <Button state="success" isLoading type="outlined">Success</Button>
  </Buttons>
</Playground>

## Outlines

An outline can be placed on a button by assigning `"outlined"` to the `type` prop.

<Playground>
  <Buttons>
    <Button state="primary" type="outlined">Primary</Button>
    <Button state="success" type="outlined">Success</Button>
    <Button state="danger" type="outlined">Danger</Button>
  </Buttons>
</Playground>

## Sizes

Buttons are available in four different sizes and can be added with a `size` prop.

<Playground>
  <Buttons>
    <Button size="small">Small</Button>
    <Button>Default</Button>
    <Button size="medium">Medium</Button>
    <Button size="large">Large</Button>
  </Buttons>
</Playground>

## Links

A button can appear as a link with no styling by adding the `isLink` prop.

<Playground>
  <Button state="primary" type="link">Default</Button>
</Playground>


## Disabled

To indicate a button as disabled, use the `disabled` prop.

<Playground>
  <Buttons>
    <Button disabled>Default</Button>
    <Button state="primary" disabled>Primary</Button>
    <Button state="success" disabled>Success</Button>
    <Button state="danger" disabled>Danger</Button>
    <Button state="warning" disabled>Warning</Button>
  </Buttons>
</Playground>

## Lists

To group a set of buttons, use the `<Buttons>` component.

<Playground>
  <Buttons>
    <Button>Previous</Button>
    <Button>Next</Button>
  </Buttons>
  <Buttons isGrouped marginTop="1rem">
    <Button>Previous</Button>
    <Button>Now</Button>
    <Button>Next</Button>
  </Buttons>
</Playground>

### Submit/Cancel buttons

## `<Button>` Props

<PropsTable of={Button} />

## `<Buttons>` Props

<PropsTable of={Buttons} />
