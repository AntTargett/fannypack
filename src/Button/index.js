export { ThemedButton as Button } from './Button';
export { ThemedButton as default } from './Button';
