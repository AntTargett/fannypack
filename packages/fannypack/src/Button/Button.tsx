import * as React from 'react';
import { Button as ReakitButton } from 'reakit';

import * as utils from '../utils';
import { BoxProps, useBoxProps } from '../Box';

import * as styles from './styles';
import { ButtonThemeConfig } from '../types';

export type LocalButtonProps = {
  overrides?: ButtonThemeConfig;
  palette: string;
  size: 'small' | 'default' | 'medium' | 'large';
};
export type ButtonProps = BoxProps & LocalButtonProps;

Button.defaultProps = {
  palette: 'default',
  size: 'default'
};

export function useButtonProps(props: Partial<ButtonProps> = {}) {
  const { className: prevClassName, ...newProps } = useBoxProps(props);
  const className = utils.useClassName({
    style: styles.Button,
    styleProps: props,
    prevClassName
  });
  return { ...newProps, className };
}

export function Button(props: ButtonProps) {
  const { children, ...restProps } = props;
  const buttonProps = useButtonProps(restProps);
  if (utils.isFunction(children)) {
    return <React.Fragment>{children(buttonProps)}</React.Fragment>;
  }
  return <ReakitButton {...buttonProps}>{children}</ReakitButton>;
}
