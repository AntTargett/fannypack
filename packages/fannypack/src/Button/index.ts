import * as buttonStyles from './styles';

export * from './Button';
export { buttonStyles };
