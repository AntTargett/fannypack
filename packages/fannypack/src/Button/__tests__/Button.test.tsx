import * as React from 'react';
import { renderHook, act } from '@testing-library/react-hooks';
import { Button, useButtonProps } from '../Button';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Button>Hello world</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <Button backgroundColor="red" color="white">
        Hello world
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with overrides', () => {
    const { container } = render(<Button overrides={{ base: { backgroundColor: 'red' } }}>Hello world</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('sizes', () => {
    it('should render a small button correctly', () => {
      const { container } = render(<Button size="small">Hello world</Button>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render a default button (no size) correctly', () => {
      const { container } = render(<Button size="default">Hello world</Button>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render a medium button correctly', () => {
      const { container } = render(<Button size="medium">Hello world</Button>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render a large button correctly', () => {
      const { container } = render(<Button size="large">Hello world</Button>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render correctly with overrides', () => {
      const { container } = render(
        <Button size="small" overrides={{ sizes: { small: { backgroundColor: 'red' } } }}>
          Hello world
        </Button>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<Button use="div">Hello world</Button>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Button props', () => {
      const { result } = renderHook(() => useButtonProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<Button>{ButtonProps => <div {...ButtonProps}>Hello world</div>}</Button>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Button.base should render correctly', () => {
    const { container } = render(<Button>hello world</Button>, {
      theme: { Button: { base: { backgroundColor: 'red' } } }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
