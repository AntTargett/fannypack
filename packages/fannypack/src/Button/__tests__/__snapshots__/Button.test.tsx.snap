// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  border: 1px solid #b3b3b3;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #444bc9 0px 0px 0px 2px;
}

<div
  class="emotion-0 emotion-1 emotion-2"
  role="button"
  tabindex="0"
  type="button"
>
  Hello world
</div>
`;

exports[`composition hook should return with Button props 1`] = `
Object {
  "as": undefined,
  "className": "css-1l676rk css-1pwrqjj css-1j9m13w",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  border: 1px solid #b3b3b3;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #444bc9 0px 0px 0px 2px;
}

<div
  class="emotion-0 emotion-1 emotion-2"
>
  Hello world
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  border: 1px solid #b3b3b3;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #444bc9 0px 0px 0px 2px;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  tabindex="0"
  type="button"
>
  Hello world
</button>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  border: 1px solid #b3b3b3;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #444bc9 0px 0px 0px 2px;
}

.emotion-2.emotion-2 {
  background-color: red;
  color: white;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  tabindex="0"
  type="button"
>
  Hello world
</button>
`;

exports[`props should render correctly with overrides 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  background-color: red;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  border: 1px solid #b3b3b3;
}

.emotion-0 {
  background-color: red;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #444bc9 0px 0px 0px 2px;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  tabindex="0"
  type="button"
>
  Hello world
</button>
`;

exports[`props sizes should render a default button (no size) correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  border: 1px solid #b3b3b3;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #444bc9 0px 0px 0px 2px;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  tabindex="0"
  type="button"
>
  Hello world
</button>
`;

exports[`props sizes should render a large button correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  border: 1px solid #b3b3b3;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #444bc9 0px 0px 0px 2px;
}

.emotion-0 {
  font-size: 1.25em;
  min-height: 3em;
  padding: 0 1.5rem;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  tabindex="0"
  type="button"
>
  Hello world
</button>
`;

exports[`props sizes should render a medium button correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  border: 1px solid #b3b3b3;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #444bc9 0px 0px 0px 2px;
}

.emotion-0 {
  min-height: 3em;
  padding: 0 1.25rem;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  tabindex="0"
  type="button"
>
  Hello world
</button>
`;

exports[`props sizes should render a small button correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  border: 1px solid #b3b3b3;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #444bc9 0px 0px 0px 2px;
}

.emotion-0 {
  font-size: 0.75em;
  min-height: 2em;
  padding: 0 0.5rem;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  tabindex="0"
  type="button"
>
  Hello world
</button>
`;

exports[`props sizes should render correctly with overrides 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  border: 1px solid #b3b3b3;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #444bc9 0px 0px 0px 2px;
}

.emotion-0 {
  font-size: 0.75em;
  min-height: 2em;
  padding: 0 0.5rem;
}

.emotion-0 {
  background-color: red;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  tabindex="0"
  type="button"
>
  Hello world
</button>
`;

exports[`theming Button.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  border: 1px solid #b3b3b3;
}

.emotion-0 {
  background-color: red;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #444bc9 0px 0px 0px 2px;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  tabindex="0"
  type="button"
>
  hello world
</button>
`;
